browser.js

```diff
@@ -3,218 +3,218 @@ window.onresize = doLayout;
-    window.bg = bg;
-    chrome.storage.local.get(null, settings_ready);
+chrome.runtime.getBackgroundPage(function (bg) {
+	window.bg = bg;
+	chrome.storage.local.get(null, settings_ready);
 })
 
 
 
-onload = function() {
-  getServerSettings(receivedServerSettings);
-  var webview = document.querySelector('webview');
-  doLayout();
-
-  document.querySelector('#reset').onclick = function() {
-    window.close();
-  };
-
-  document.querySelector('#reload').onclick = function() {
-    if (isLoading) {
-      webview.stop();
-    } else {
-      webview.reload();
-    }
-  };
-  document.querySelector('#reload').addEventListener(
-    'webkitAnimationIteration',
-    function() {
-      if (!isLoading) {
-        document.body.classList.remove('loading');
-      }
-    });
-
-
-  webview.addEventListener('exit', handleExit);
-  webview.addEventListener('loadstart', handleLoadStart);
-  webview.addEventListener('loadstop', handleLoadStop);
-  webview.addEventListener('loadabort', handleLoadAbort);
-  webview.addEventListener('loadredirect', handleLoadRedirect);
-  webview.addEventListener('loadcommit', handleLoadCommit);
-  webview.addEventListener('permissionrequest', handleRequest);
+onload = function () {
+	getServerSettings(receivedServerSettings);
+	var webview = document.querySelector('webview');
+	doLayout();
+
+	document.querySelector('#reset').onclick = function () {
+		window.close();
+	};
+
+	document.querySelector('#reload').onclick = function () {
+		if (isLoading) {
+			webview.stop();
+		} else {
+			webview.reload();
+		}
+	};
+	document.querySelector('#reload').addEventListener(
+		'webkitAnimationIteration',
+		function () {
+			if (!isLoading) {
+				document.body.classList.remove('loading');
+			}
+		});
+
+
+	webview.addEventListener('exit', handleExit);
+	webview.addEventListener('loadstart', handleLoadStart);
+	webview.addEventListener('loadstop', handleLoadStop);
+	webview.addEventListener('loadabort', handleLoadAbort);
+	webview.addEventListener('loadredirect', handleLoadRedirect);
+	webview.addEventListener('loadcommit', handleLoadCommit);
+	webview.addEventListener('permissionrequest', handleRequest);
 };
 
 
 function getServerSettings(callback) {
-	chrome.storage.managed.get("serverSettings",function(results) {
+	chrome.storage.managed.get("serverSettings", function (results) {
 		if (chrome.runtime.lastError) {
 			console.log("error, returning empty. Error Message: " + chrome.runtime.lastError.message);
 			return;
 		} else {
 			console.log("got server settings, returning");
 			callback(results.serverSettings)
-			
+
 		}
 	});
 }
 
 function receivedServerSettings(serverSettings) {
 	console.log("got server settings");
 	console.log(serverSettings);
-	var url = convertServerSettingsToUrl(serverSettings);
-	
-	if (url && url.length > 0) {
-		console.log("calling navigateTo url: " + url);
-		navigateTo(url);
-	}
-	
+	chrome.storage.local.get('relativeLaunchUrl', function (obj) {
```

Comment:
Check to see if a launch URL was stored in local storage, so we can combine its parameters with the server settings stored in managed storage.